wallet/src/main/java/com/android/identity_credential/wallet/PermissionTracker.kt

```diff
@@ -60,19 +61,30 @@ class PermissionTracker(private val activity: ComponentActivity, private val per
      */
     @Composable
     fun PermissionCheck(permissions: Iterable<String>, content: @Composable () -> Unit) {
-
         var allGranted = true
+        val permissionsToRequest: MutableList<String> = mutableListOf()
         for (permission in permissions) {
             if (!mState[permission]!!.observeAsState().value!!) {
                 allGranted = false
-                Text(this.permissions[permission]!!, modifier = Modifier.fillMaxWidth())
-                Button(onClick = {requestPermission(permission)}) {
-                    Text("Request")
-                }
+                permissionsToRequest.add(permission)
             }
         }
+
         if (allGranted) {
             content()
+        } else {
+            SinglePermissionRequest(permission = permissionsToRequest[0])
+        }
+    }
+
+    @Composable
+    private fun SinglePermissionRequest(permission: String){
+        val reasoningTxt: String = this.permissions[permission]!!
+        Column {
+            Text(text = reasoningTxt, modifier = Modifier.fillMaxWidth())
+            Button(onClick = {requestPermission(permission)}) {
+                Text("I understand")
+            }
         }
```

Comment:
CC @sorotokin to review the changes here.

Reply:
I am not sure why 
```
} else {
    SinglePermissionRequest(permission = permissionsToRequest[0])
}
```
is needed? It should give you a list of all the missing permission with a button to request each of them. Is it not working?